lib/cocoapods/generator/xcconfig/pod_xcconfig.rb

```diff
@@ -72,8 +72,8 @@ def generate
           @xcconfig.merge! XCConfigHelper.settings_for_dependent_targets(target, target.recursive_dependent_targets)
           if @test_xcconfig
             @xcconfig.merge! XCConfigHelper.settings_for_dependent_targets(target, target.test_dependent_targets)
-            XCConfigHelper.generate_vendored_build_settings(nil, target.test_dependent_targets, @xcconfig)
-            XCConfigHelper.generate_other_ld_flags(nil, target.test_dependent_targets, @xcconfig)
+            XCConfigHelper.generate_vendored_build_settings(nil, [target, *target.test_dependent_targets], @xcconfig)
```

Comment:
we no longer manually inject the target itself as a `test_dependent_target` so this will need to be added manually when we wire up the xcodeproj.

tests still pass, no need for a new one.